test/grub2_test.rb

```diff
@@ -0,0 +1,159 @@
+
+      subject.write
+    end
+
+    it "changes pmbr flag as specified in pmbr_action for all boot devices with gpt label" do
+      stage1 = double(Bootloader::Stage1, model: double(devices: ["/dev/sda", "/dev/sdb1"], generic_mbr?: false), write: nil)
+      allow(Bootloader::Stage1).to receive(:new).and_return(stage1)
+
+      allow(Yast::Storage).to receive(:GetDisk) do |m, dev|
+        case dev
+        when "/dev/sda" then { "device" => "/dev/sda", "label" => "msdos" }
+        when "/dev/sdb1" then { "device" => "/dev/sdb", "label" => "gpt" }
+        else raise "unknown device #{dev}"
+        end
+      end
+
+      expect(subject).to receive(:pmbr_setup).with("/dev/sdb")
+
+      subject.write
+    end
+
+    it "runs grub2-install for all configured stage1 locations" do
+      stage1 = double(Bootloader::Stage1, model: double(devices: ["/dev/sda", "/dev/sdb1"], generic_mbr?: false), write: nil)
+      allow(Bootloader::Stage1).to receive(:new).and_return(stage1)
+
+      grub2_install = double(Bootloader::GrubInstall)
+      expect(grub2_install).to receive(:execute).with(devices: ["/dev/sda", "/dev/sdb1"])
+      expect(Bootloader::GrubInstall).to receive(:new).with(efi: false).and_return(grub2_install)
+
+      subject.write
+    end
+
+    context "on s390" do
+      before do
+        allow(Yast::Arch).to receive(:architecture).and_return("s390_64")
+      end
+
+      it "does not run mbr update for configured stage1 flags" do
+        expect(Bootloader::MBRUpdate).to_not receive(:new)
+
+        subject.write
+      end
+    end
+
+    context "on other architectures" do
+      before do
+        allow(Yast::Arch).to receive(:architecture).and_return("x86_64")
+      end
+
+      it "runs mbr update for configured stage1 flags" do
+        mbr_update = double(Bootloader::MBRUpdate)
+        expect(mbr_update).to receive(:run)
+        expect(Bootloader::MBRUpdate).to receive(:new).and_return(mbr_update)
+
+        subject.write
+      end
+    end
+  end
+
+  context "#propose" do
+    before do
+      stage1 = double.as_null_object
+      allow(Bootloader::Stage1).to receive(:new).and_return(stage1)
+    end
+
+    it "proposes stage1" do
+      stage1 = double
+      expect(stage1).to receive(:propose)
+      allow(Bootloader::Stage1).to receive(:new).and_return(stage1)
+
+      subject.propose
+    end
+
+    it "propose to remove pmbr flag if boot disk is with gpt label" do
+      allow(Yast::BootStorage).to receive(:gpt_boot_disk?).and_return(true)
+
+      subject.propose
+
+      expect(subject.pmbr_action).to eq :remove
+    end
+  end
+
+  describe "#name" do
+    it "returns \"grub2\"" do
+      expect(subject.name).to eq "grub2"
+    end
+  end
+
+  describe "#packages" do
+    it "return list containing grub2 package" do
+      expect(subject.packages).to include("grub2")
+    end
+
+    it "returns list containing syslinux package if generic_mbr is used" do
```

Comment:
I can live with the current state, so I would give a LGTM if you insist, but for every situation in which we have to decide about testing POV vs documentation POV I would always favor the first one.

We use RSpec primary for having unit tests and secondary for documentation purposes. Even of the output gets longer I prefer to always have the complementary scenarios. On the other hand longer does not necessarily means worst.